controllers/osbuild_controller_test.go

```diff
@@ -0,0 +1,422 @@
+			},
+			JSON200: &composer.ComposeStatus{
+				Id: zeroUuid,
+				ImageStatus: composer.ImageStatus{
+					Status: composer.ImageStatusValueBuilding,
+				},
+				Status: composer.ComposeStatusValuePending,
+			},
+		}
+	})
+
+	AfterEach(func() {
+		mockCtrl.Finish()
+		osbuildInstance.DeletionTimestamp = nil
+		osbuildInstance.Status.Conditions = nil
+		osbuildInstance.Status.ContainerComposeId = controllers.EmptyComposeID
+	})
+
+	Context("Failure to get OSBuild instance", func() {
+		It("Should return Done when the instance is not found", func() {
+			// given
+			osBuildRepository.EXPECT().Read(requestContext, instanceName, instanceNamespace).Return(nil, errNotFound)
+			// when
+			result, err := reconciler.Reconcile(requestContext, request)
+			// then
+			Expect(err).To(BeNil())
+			Expect(result).To(Equal(resultDone))
+
+		})
+
+		It("Should return error when failed to get the instance", func() {
+			// given
+			osBuildRepository.EXPECT().Read(requestContext, instanceName, instanceNamespace).Return(nil, errFailed)
+			// when
+			result, err := reconciler.Reconcile(requestContext, request)
+			// then
+			Expect(err).To(BeNil())
+			Expect(result).To(Equal(resultShortRequeue))
+		})
+	})
+
+	Context("Handle deletion", func() {
+		It("Should return Done if removed the finalizer successfully", func() {
+			// given
+			osbuildInstance.DeletionTimestamp = &metav1.Time{Time: time.Now()}
+			osBuildRepository.EXPECT().Read(requestContext, instanceName, instanceNamespace).Return(&osbuildInstance, nil)
+			// when
+			result, err := reconciler.Reconcile(requestContext, request)
+			// then
+			Expect(err).To(BeNil())
+			Expect(result).To(Equal(resultDone))
+		})
+	})
+
+	Context("Create edge-container request ", func() {
+		It("and fail on postCompose with error", func() {
+			// given
+			osBuildRepository.EXPECT().Read(requestContext, instanceName, instanceNamespace).Return(&osbuildInstance, nil)
+			osBuildRepository.EXPECT().PatchStatus(requestContext, gomock.Any(), gomock.Any()).Return(nil)
+			composerClient.EXPECT().PostComposeWithResponse(requestContext, gomock.Any()).Return(nil, errFailed)
+			// when
+			result, err := reconciler.Reconcile(requestContext, request)
+			// then
+			Expect(err).To(BeNil())
+			Expect(result).To(Equal(resultLongRequeue))
+		})
+		It("and on postCompose returns status code bad request ", func() {
+			// given
+			osBuildRepository.EXPECT().Read(requestContext, instanceName, instanceNamespace).Return(&osbuildInstance, nil)
+			osBuildRepository.EXPECT().PatchStatus(requestContext, gomock.Any(), gomock.Any()).Return(nil)
+			composerClient.EXPECT().PostComposeWithResponse(requestContext, gomock.Any()).Return(&composerPostResponseFailed, nil)
+
+			// when
+			result, err := reconciler.Reconcile(requestContext, request)
+			// then
+			Expect(err).To(BeNil())
+			Expect(result).To(Equal(resultLongRequeue))
+		})
+		It("and on postCompose returns status code created ", func() {
```

Comment:
Same here. Try reading it "It <description>". That's why in most cases the description starts with `should`.

Reply:
updated it please review